src/app/favourites/_components/FavElement.tsx

```diff
@@ -0,0 +1,86 @@
+'use client'
+
+import Image from 'next/image'
+import productImg from '../../../../public/coffee.png'
+import { IProduct } from '@/models/Products'
+
+import { useCombinedStore } from '@/store/store'
+import { useFavouritesStore } from '@/store/favStore'
+import AddToCartButton from '@/components/ui/Button'
+import ButtonHeart from '@/components/Heart/ButtonHeart'
+
+interface FavElementProps {
+  product: IProduct
+}
+
+export default function FavElement({ product }: FavElementProps) {
+  const addToCart = useCombinedStore((state) => state.add)
+  const { favouriteIds, addFavourite, removeFavourite } = useFavouritesStore()
+
+  const isFavourite =
+    favouriteIds && favouriteIds.some((favId) => favId === product.id)
+
+  const handleAddToCart = () => {
+    addToCart({
+      id: product.id,
+      info: {
+        name: product.name,
+        price: product.price,
+        description: product.description,
+        active: product.active,
+        quantity: product.quantity,
+      },
+      quantity: 1,
+    })
+  }
+
+  if (!isFavourite) {
+    addFavourite(product.id)
+  } else {
+    removeFavourite(product.id)
+  }
+
+  const handleToggleFavourite = () => {
+    try {
+      if (!isFavourite) {
+        addFavourite(product.id)
+      } else {
+        removeFavourite(product.id)
+      }
+    } catch (error) {
+      console.error('Error toggling favourite:', error)
+    }
+  }
```

Comment:
That should bee just remove from favourite list function. And u have that function in the store, thus that function is unnecessary

Reply:
Fixed